src/pages/guides/getting-started/writing-a-file/index.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Writing a file with UXP
+description: Interact with the file system and other operating system operations using the UXP APIs in your Photoshop plugin
+contributors:
+  - https://github.com/pklaschka
+---
+
+# Writing a file with UXP
+
+In this tutorial, you will enhance your plugin by adding a button to export a `.tsv` file that lists all your document's
+layers including some details.
+
+## Prerequisites
+
+This tutorial builds on top of the plugin from [the previous tutorial](../editing-the-document/index.md).
+
+## Step 1: Adjust the `index.html`
+
+In this tutorial, we want to add a new function to the plugin: Exporting a `.tsv` file.
+
+To support this in your UI, adjust your `index.html` file to include another section with another button `<sp-button id="btnExport" />`:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <script src="index.js"></script>
+  </head>
+  <style>
+    body {
+      color: white;
+      padding: 0 16px;
+    }
+  </style>
+  <body>
+    <sp-heading>Layers</sp-heading>
+    <sp-heading size="S">Rename layers</sp-heading>
+    <sp-body size="S"> Rename layers to contain their opacity. </sp-body>
+    <sp-button id="btnRename">Rename Layers</sp-button>
+    <sp-heading size="S">Export Report</sp-heading>
+    <sp-body size="S"> Export a report of all layers in the document. </sp-body>
+    <sp-button id="btnExport" variant="secondary">Export Report</sp-button>
+  </body>
+</html>
+```
+
+The resulting panel looks like this:
+
+![Photoshop panel with two buttons: "Rename Layers" and "Export Report"](panel.png)
+
+## Step 2: Add the new function to the `index.js`
+
+To export the file, you need to do two things:
+
+1. create a string with the desired content of the file. This works the same as in the previous tutorials using the DOM APIs from the `window.require('photoshop')` module.
+2. Use the UXP APIs, which you can access using `window.require('uxp')`, to ask the user for a location to save the file and save a file with the desired content to that location.
+
+Add the following code to your existing `index.js` file:
+
+```js
+// [...]
+
+async function exportReport() {
+    // create the TSV string
+    let tsvString = "Base name\tOpacity\tIs visible";
+
+    const app = window.require("photoshop").app;
+    app.activeDocument.layers.forEach((layer) => {
+        tsvString +=
+            "\n" +
```

Comment:
Could we use string literals here instead of concatenation?

Reply:
Meaning `tsvString += `\n${layer.name}\t${layer.opacity}`...? 


But this could be easier to read.... up to you.

Reply:
I originally used a string literal but decided against it for readability reasons (the line got quite long that way) 😃